File: solutions/2788/main.py
from typing import * 
from itertools import chain

class Solution:
    def splitWordsBySeparator(self, words: List[str], separator: str) -> List[str]:
        words = [w.split(separator) for w in words]
        # flatten the list 
        words = list(chain(*words))
        words = [w for w in words if len(w) > 0]
        return words 
    
if __name__ == '__main__': 
    s = Solution()
    
    assert s.splitWordsBySeparator(["one.two.three","four.five","six"], ".") == ["one","two","three","four","five","six"]

File: solutions/2788/main.rs
struct Solution {}

impl Solution {
    pub fn split_words_by_separator(words: Vec<String>, separator: char) -> Vec<String> {
        words
            .iter()
            .flat_map(|s| s.split(separator).collect::<Vec<_>>())
            .map(|s| s.to_string() )
            .filter(|s| !s.is_empty() )
            .collect::<Vec<_>>()
    }
}

fn main() {
    Solution::split_words_by_separator(
        vec![
            "one.two.three".to_string(),
            "four.five".to_string(),
            "six".to_string(),
        ],
        '.',
    );
}
